server/src/main/java/org/opensearch/tracing/TracerManager.java

```diff
@@ -0,0 +1,116 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.tracing.noop.NoopTracer;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+/**
+ * TracerManager represents a single global class that is used to access tracers.
+ *
+ * The Tracer singleton object can be retrieved using TracerManager.getTracer(). The TracerManager object
+ * is created during class initialization and cannot subsequently be changed.
+ */
+public class TracerManager {
+
+    private static final Logger logger = LogManager.getLogger(TracerManager.class);
+    private static volatile TracerManager INSTANCE;
+
+    private volatile Tracer defaultTracer;
+    private final Object mutex = new Object();
+    private final TracerSettings tracerSettings;
+    private final Supplier<Telemetry> telemetrySupplier;
+    private final ThreadPool threadPool;
+
+    /**
+     * Initializes the TracerFactory singleton instance
+     *
+     * @param tracerSettings       tracer settings instance
+     * @param threadPool           thread pool instance
+     */
+    public static synchronized void initTracerManager(
+        TracerSettings tracerSettings,
+        Supplier<Telemetry> tracerSupplier,
+        ThreadPool threadPool
+    ) {
+        if (INSTANCE == null) {
+            INSTANCE = new TracerManager(tracerSettings, tracerSupplier, threadPool);
+        } else {
+            logger.warn("Trying to double initialize TracerFactory, skipping");
+        }
+    }
+
+    /**
+     * Returns the {@link Tracer} singleton instance
+     * @return Tracer instance
+     */
+    public static Tracer getTracer() {
+        return INSTANCE == null ? NoopTracer.INSTANCE : INSTANCE.tracer();
+    }
+
+    public static BiConsumer<Map<String, String>, Map<String, Object>> getTracerHeaderInjector() {
+        return INSTANCE == null ? (x, y) -> {} : INSTANCE.tracerHeaderInjector();
+    }
+
+    /**
+     * Closes the {@link Tracer}
+     */
+    public static void closeTracer() {
+        if (INSTANCE != null && INSTANCE.defaultTracer != null) {
+            try {
+                INSTANCE.defaultTracer.close();
+            } catch (IOException e) {
+                logger.warn("Error closing tracer", e);
+            }
+        }
+    }
+
+    public TracerManager(TracerSettings tracerSettings, Supplier<Telemetry> telemetrySupplier, ThreadPool threadPool) {
+        this.tracerSettings = tracerSettings;
+        this.telemetrySupplier = telemetrySupplier;
+        this.threadPool = threadPool;
+    }
+
+    private Tracer tracer() {
+        return isTracingDisabled() ? NoopTracer.INSTANCE : getOrCreateDefaultTracerInstance();
+    }
```

Comment:
This should probably go in initTracer directly

Reply:
I believe you mean static `getTracer` method. The check in this method is primarily to check if the TracerManager is not initialized or not. Returning the tracer based on the run time bindings seems to be a responsibility of Tracer manager hence i kept in the non-static `tracer()` method

Reply:
This code is removed now